src/qibo/tests/test_core_gates.py

```diff
@@ -65,6 +65,46 @@ def test_z(backend):
     K.assert_allclose(final_state, target_state)
 
 
+def test_s(backend):
+    final_state = apply_gates([gates.H(0), gates.H(1), gates.S(1)], nqubits=2)
+    target_state = np.zeros_like(final_state)
+    target_state[0] = 0.5
+    target_state[1] = 0.5j
+    target_state[2] = 0.5
+    target_state[3] = 0.5j
```

Comment:
Here and in the tests below, since we modify all elements, we can just define the target vector as:
```Python
target_state = np.array([0.5, 0.5j, 0.5, 0.5j])
```
I believe it is simpler this way, but is okay as it is anyway.